pkg/argocd/argocd.go

```diff
@@ -425,15 +434,17 @@ func SetHelmImage(app *v1alpha1.Application, newImage *image.ContainerImage) err
 		}
 	}
 
-	if app.Spec.Source.Helm == nil {
-		app.Spec.Source.Helm = &v1alpha1.ApplicationSourceHelm{}
+	// This case seems not possible, but we check it anyway
+	helmSpec := app.Spec.GetSource().Helm
```

Comment:
I change some code back to the original to keep the same syntax as the rest of the code. Also to initialize the app.Spec.Source.Helm value. This comes from the changes to fix the error in unit tests from the Kustomize section below.

```suggestion
	app.Spec.Source.Helm = app.Spec.GetSource().Helm
	if app.Spec.Source.Helm == nil {
		app.Spec.Source.Helm = &v1alpha1.ApplicationSourceHelm{}
	}

	if app.Spec.Source.Helm.Parameters == nil {
		app.Spec.Source.Helm.Parameters = make([]v1alpha1.HelmParameter, 0)
	}

	app.Spec.Source.Helm.Parameters = mergeHelmParams(app.Spec.Source.Helm.Parameters, mergeParams)

	return nil
```

Reply:
I added this two snippets to fix the errors showed by the unit tests and to keep the same code patterns on the Helm and Kustomize sections.

Please, let me know if this works for you in order to merge this fix and solve the problem with multisource applications as we are experiencing the same issues and this fix them. Thank you!